### PR TITLE
TextFileInput changes

### PR DESCRIPTION
**Pull request by AlexanderBuloichik**

### CODE REVIEW

File: engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjection.java [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+import org.pentaho.di.trans.step.StepMetaInjectionInterface;
+
+/**
+ * This takes care of the external metadata injection into the TextFileInputMeta class
+ *
+ * @author Matt
+ */
+public class TextFileInputMetaInjection implements StepMetaInjectionInterface {
+
+  enum Entry implements StepMetaInjectionEntryInterface {
```
Comment (deinspanjer) [inline engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjection.java]: The indentation on this enum went crazy.

File: engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputUtils.java [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+      return line.toString();
+    }
+    if ( line.length() > 0 ) {
+      return line.toString();
+    }
+
+    return null;
+  }
+
+  public static final Object[] convertLineToRow( LogChannelInterface log, TextFileLine NewFileLine,
```
Comment (deinspanjer) [inline engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputUtils.java]: A couple of argument variables here have bad names:
s/NewFileLine/textFileLine
s/additionalOutputFIelds/additionalOutputFields

**Comment by deinspanjer** [comment]
I would still prefer the prefix BaseInput to be changed to BaseFileInput.  Without the word File, a dev *might* be confused into thinking that one of these classes should be used as a base for some non-file related input step.

I won't put my foot down, because it is a mistake that would quickly be found when they tried to extend it and suddenly had to deal with all sorts of methods that wouldn't apply to them, but still...

**Comment by bkemper-pentaho** [comment]
Put your foot down.   You're correct.  Please do the refactor prior to merge...

Bob Kemper
Executive Vice President of Engineering
Pentaho, a Hitachi Data Systems company.
5950 Hazeltine National Dr.
Suite 340
Orlando, FL 32822
(407)-965-4019

[cid:73B6094B-B76D-497B-8724-CECB1ECA14D5]

File: engine/src/org/pentaho/di/trans/steps/fileinput/text/InputFileMetaInterface.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.fileinput.text;
+
+import org.pentaho.di.core.variables.VariableSpace;
+import org.pentaho.di.trans.step.StepMetaInterface;
+
+public interface InputFileMetaInterface extends StepMetaInterface {
```
Comment (deinspanjer) [inline engine/src/org/pentaho/di/trans/steps/fileinput/text/InputFileMetaInterface.java]: This interface is not used in the new step/package.  Why should we keep it?

**Comment by deinspanjer** [comment]
Okay, that is all the pertinent feedback I have from my exhaustive comparison of the old vs refactored code.

File: engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjection.java (hunk 1 above)
Comment (AlexanderBuloichik) [inline engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjection.java]: Yes, it happen after Eclipse's formatting. Looks like Eclipse's formatting settings don't 100% correrspond with checkstyle settings.

**Comment by AlexanderBuloichik** [comment]
Closed because new one was pushed (#1716)

**Pull request closed without merging**